tests/sagemaker/mock/__init__.py

```diff
@@ -473,6 +492,7 @@ def delete_endpoint(self, endpoint_name):
 
         del self.endpoints[endpoint_name]
 
+    @paginate(pagination_model=PAGINATION_MODEL)
```

Comment:
I used this decorator from `moto` and added the configuration to `PAGINATION_MODEL` - https://github.com/getmoto/moto/blob/4.2.14/moto/sagemaker/models.py#L1982. This could be extended to other methods that utilize pagination.

This decorator operates on the list returned by `sagemaker_backend.list_endpoints` and truncates according to `MaxResults` and allows the `NextToken` to be returned where applicable.

For `moto==5.x.x`, `list_endpoints` is [mocked natively](https://github.com/getmoto/moto/blob/master/moto/sagemaker/models.py#L98).